File: c++/14_Longest_Common_Prefix.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans;
        char c;
        for(int i=0; i<strs[0].size();i++)
        {
            c = strs[0][i];
            for(auto s:strs)
            {
                if( (i+1)>s.size() || c!=s[i] )
                    return ans;

            }
            ans.push_back(c);
        }
        return ans;
    }
};

int main()
{
    vector<string> strs = {"dog","racecar","car"};
    Solution s;
    string ans = s.longestCommonPrefix(strs);
    /*
    for(auto s : strs)
    {
        printf("%d\n",s.size());
    }
    */
    printf("%s",ans.c_str());

    return 0;    
}

File: python/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        compare = strs[0]
        ans = ""

        min_len = 201
        for s in strs:
            if len(s) < min_len:
                min_len = len(s)

        for i in range(min_len):
            is_common = True
            for s in strs[1:]:
                if s[i] == compare[i]:
                    continue
                else:
                    is_common = False
                    break
            if is_common:
                ans += compare[i]
            else:
                break

        return ans


if __name__ == "__main__":
    strs = ["flower", "flow", "flight"]
    strs = sorted(strs)
    print(f"min : {strs}")
    ans = "fl"
    output = Solution().longestCommonPrefix(strs)
    print(output)
